corehq/apps/app_manager/util.py

```diff
@@ -408,64 +408,68 @@ def get_casedb_schema(form):
 
     This lists all case types and their properties for the given app.
     """
-    app = form.get_app()
-    base_case_type = form.get_module().case_type
-    case_types = app.get_case_types() | get_shared_case_types(app)
-    per_type_defaults = get_per_type_defaults(app.domain, case_types)
-    builder = ParentCasePropertyBuilder(app, ['case_name'], per_type_defaults)
-    related = builder.get_parent_type_map(case_types, allow_multiple_parents=True)
-    map = builder.get_case_property_map(case_types, include_parent_properties=False)
-    descriptions_dict = get_case_property_description_dict(app.domain)
-
-    if base_case_type:
-        # Generate hierarchy of case types, represented as a list of lists of strings:
-        # [[base_case_type], [parent_type1, parent_type2...], [grandparent_type1, grandparent_type2...]]
-        # Vellum case management only supports three levels
-        generation_names = ['case', 'parent', 'grandparent']
-        generations = [[] for g in generation_names]
-
-        def _add_ancestors(ctype, generation):
-            if generation < len(generation_names):
-                generations[generation].append(ctype)
-                for parent in related.get(ctype, {}).get('parent', []):
-                    _add_ancestors(parent, generation + 1)
-
-        _add_ancestors(base_case_type, 0)
-
-        # Remove any duplicate types or empty generations
-        generations = [set(g) for g in generations if len(g)]
-    else:
-        generations = []
-
-    subsets = [{
-        "id": generation_names[i],
-        "name": "{} ({})".format(generation_names[i], " or ".join(ctypes)) if i > 0 else base_case_type,
-        "structure": {
-            p: {"description": descriptions_dict.get(t, {}).get(p, '')}
-            for t in ctypes for p in map[t]},
-        "related": {"parent": {
-            "hashtag": "#case/" + generation_names[i + 1],
-            "subset": generation_names[i + 1],
-            "key": "@case_id",
-        }} if i < len(generations) - 1 else None,
-    } for i, ctypes in enumerate(generations)]
+    @quickcache(['app.id', 'app.version'])
+    def _inner(app, form):
```

Comment:
Do you want this to be cached across different calls to `get_casedb_schema`? I've never tried the decorator on an inner function like this, and I can't see why it wouldn't work—the cache key should still match up. (`@memoized` definitely would not work here, since its cache is stored on the function object itself, which is what made this catch my attention.

Anyway I think this is fine despite my initial side-glance, especially if you tried it and it's clearly caching.